css-view-transitions-2/Overview.bs

```diff
@@ -264,6 +281,34 @@ Note: this event is fired when [=reveal document|revealing a document=].
 The <dfn attribute for=PageRevealEvent>viewTransition</dfn> [=getter steps=] are to return the
 [=inbound cross-document view-transition=] for [=this's=] [=relevant global object's=] [=associated document=].
 
+## Additions to {{Document}} ## {#additions-to-document-api}
+
+	<xmp class=idl>
+		dictionary StartViewTransitionOptions {
+			UpdateCallback? update = null;
+			sequence<DOMString>? types = null;
+		};
+
+		partial interface Document {
+
+			ViewTransition startViewTransition(optional (UpdateCallback or StartViewTransitionOptions)? callbackOrOptions = null);
+		};
+	</xmp>
+
+### {{Document/startViewTransition(options)}} Method Steps ### {#ViewTransition-start-with-options}
+
+	<div algorithm="start-vt-with-options">
+		The [=method steps=] for <dfn method for=Document>startViewTransition(|callbackOrOptions|)</dfn> are as follows:
+
+		1. If |callbackOrOptions| is an {{UpdateCallback}}, then run the [=method steps=] for {{Document/startViewTransition(updateCallback)}} given |callbackOrOptions| and return the result.
+
+		1. Let |viewTransition| be the [=method steps=] for {{Document/startViewTransition(updateCallback)}} given |callbackOrOptions|'s {{StartViewTransitionOptions/update}}.
```

Comment:
The above step says "run the method steps" while this one says, "Let |viewTransition| be the method steps for ...". Why the difference? I was expecting that we'd run the method steps unconditionally and = set the active types (if there is any) after that.

Reply:
It's equivalent to the following code:
```js
if (typeof callbackOrOptions === "function")
  return startViewTransitonInternal(callback);

const viewTransition = startViewTransitionInternal(callbackOrOptions.update);
viewTransition.activeTypes = callbackOrOptions.types;
return viewTransition;
```

Sure we could do something like:
```js
const update = typeof callbackOrOptions === "function" ? callbackOrOptions : callbackOrOptions.update;
const viewTransition = startViewTransitionInternal(update);
if (callbackOrOptions?.types)
  viewTransition = callbackOrOptions.types;
```

But I'm not sure it's an improvement.

Reply:
Ack. This was probably related to the comment below. I incorrectly assumed the transition types are a global state, so we should be updating it if a second transition runs with a null types list. But since its stored on the transition itself, it doesn't matter.